xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/classes/PageClassPropertyValuesProvider.java

```diff
@@ -93,24 +93,31 @@ protected PropertyValues getAllowedValues(PageClass pageClass, int limit, String
         return icon;
     }
 
-    @Override
-    protected String getLabel(DocumentReference documentReference, Object currentLabel)
+    private String getLabel(EntityReference entityReference)
     {
-        String label = currentLabel == null ? "" : currentLabel.toString().trim();
-        if (label.isEmpty()) {
-            try {
-                XWikiContext xcontext = this.xcontextProvider.get();
-                XWikiDocument document = xcontext.getWiki().getDocument(documentReference, xcontext);
-                label = document.getRenderedTitle(Syntax.PLAIN_1_0, xcontext);
-            } catch (XWikiException e) {
-                this.logger.error("Error while loading the document [{}]. Root cause is [{}]", documentReference,
-                    ExceptionUtils.getRootCause(e));
-                label = super.getLabel(documentReference, currentLabel);
+        String label;
+        try {
+            XWikiContext xcontext = this.xcontextProvider.get();
+            XWikiDocument document = xcontext.getWiki().getDocument(entityReference, xcontext);
+            label = document.getRenderedTitle(Syntax.PLAIN_1_0, xcontext);
+        } catch (XWikiException e) {
+            this.logger.error("Error while loading the document [{}]. Root cause is [{}]", entityReference,
+                ExceptionUtils.getRootCause(e));
+            if (entityReference instanceof DocumentReference) {
```

Comment:
I think the common practice is, in general, to check

    entityReference.type == EntityType.DOCUMENT

and then use

    new DocumentReference(entityReference)

but what you did will work fine in this context.